### PR TITLE
Js Support

### PR DESCRIPTION
**Pull request by KevinSchildhorn**
## Web Support

This PR adds Web support for Droidcon. 
> Note: This PR uses JS, not WASM, because of certain limitations of libraries. When those libraries are updated we can include WASM Support.

> Note: The UI is not final, hence the draft. We are using Adaptive Layout and need to adjust some small parts

How to run:
`./gradlew web:jsBrowserDevelopmentRun`

To test local times, you need to set `testNotificationTimes = true` in `shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultSyncService.kt`

### Screenshots

I have to fix these

| Android | Web |
| - | - |
| Screenshot 2026-04-22 at 1 53 14 PM | Screenshot_20260422_141042 |
| Screenshot 2026-04-22 at 2 11 00 PM |  |
| Screenshot 2026-04-22 at 2 11 31 PM | Screenshot_20260422_141054 |
| Screenshot 2026-04-22 at 2 12 38 PM | Screenshot_20260422_141133 |
| Screenshot 2026-04-22 at 2 13 32 PM | Screenshot_20260422_141339 |
| Screenshot 2026-04-22 at 2 13 50 PM | Screenshot_20260422_141352 |
| Screenshot 2026-04-22 at 2 14 08 PM | Screenshot_20260422_141401 |
| Screenshot 2026-04-22 at 2 14 12 PM | Screenshot_20260422_141415 |
| Screenshot 2026-04-22 at 2 15 14 PM | Screenshot_20260422_141511 |
| Screenshot 2026-04-22 at 2 17 15 PM | Screenshot_20260422_141649 |

### CODE REVIEW

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/gateway/impl/DefaultSponsorGateway.kt [hunk 1]
```diff
@@ -18,19 +22,50 @@ class DefaultSponsorGateway(
...
+                    groups.map { group ->
+                        log.i { "Found a Group of Sponsors" }
+                        SponsorGroupWithSponsors(
+                            group,
+                            sponsorRepository.allByGroupName(group.name, id),
+                        )
+                    }
+                }
+            }
+/*
```
Comment (wellingtoncosta) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/gateway/impl/DefaultSponsorGateway.kt]: Can we remove this dead code?

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        sponsorGroupTableAdapter = SponsorGroupTable.Adapter(
+            intToLongAdapter,
+        ),
+        conferenceTableAdapter = ConferenceTable.Adapter(
+            conferenceTimeZoneAdapter = timeZoneAdapter,
+            // Note: selectedAdapter will be added when the adapter is regenerated
+        ),
+    )
+
+    return initKoin(
```
Comment (wellingtoncosta) [inline web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: Philosophical discussion: should all dependencies be a `single`?

File: build.gradle.kts [hunk 3]
```diff
@@ -45,7 +45,8 @@ subprojects {
         }
     }
 }
-
+/*
```
Comment (wellingtoncosta) [inline build.gradle.kts]: Can be removed?

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt [hunk 4]
```diff
@@ -5,14 +5,14 @@ import kotlinx.coroutines.flow.Flow
...
     fun getProjectId(): String
-    fun getCollectionName(): String
-    fun getApiKey(): String
-    fun getScheduleId(): String
-    fun showVenueMap(): Boolean
-    fun observeChanges(): Flow<Conference>
+    suspend fun getCollectionName(): String
+    suspend fun getApiKey(): String
+    suspend fun getScheduleId(): String
+    suspend fun showVenueMap(): Boolean
```
Comment (wellingtoncosta) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt]: Why don't we use the `showVenueMap` declared in `Conference`?

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt [hunk 5]
```diff
@@ -5,14 +5,14 @@ import kotlinx.coroutines.flow.Flow
 import kotlinx.datetime.TimeZone
 
 interface ConferenceConfigProvider {
```
Comment (wellingtoncosta) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt:7]: Could we expose the domain entity instead of fragmenting it into methods that are like a getter?

```kotlin
suspend fun getCurrent(): Conference
fun observeChanges(): Flow<Conference>
```

Looking at the `DefaultConferenceConfigProvider`, all the methods are basically a "proxy" of the actual `currentConference`.

By the way, this way it'd look more like a Repository than a Provider, which is kinda vague concept in domain module.

**Review by wellingtoncosta — COMMENTED** [review]
Left some comments. LGTM overall.

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt (hunk 4 above)
Comment (KevinSchildhorn) [reply shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt]: Actually it looks like this function isn't removed anywhere, I've removed it.

File: build.gradle.kts (hunk 3 above)
Comment (KevinSchildhorn) [reply build.gradle.kts]: removed

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/gateway/impl/DefaultSponsorGateway.kt (hunk 1 above)
Comment (KevinSchildhorn) [reply shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/gateway/impl/DefaultSponsorGateway.kt]: removed

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt (hunk 5 above)
Comment (KevinSchildhorn) [reply shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/ConferenceConfigProvider.kt:7]: We could potentially do that, but it would be a somewhat large refactor and it doesn't affect the web support so I'd rather handle that at a later time.

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt (hunk 2 above)
Comment (KevinSchildhorn) [reply web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: That's a fair question. Some should definitely be singles such as the Database or Settings, ResourceReader could be anything. I don't think there's a downside to them being singles? None of them need scope or need a fresh item

**Review by wellingtoncosta — APPROVED** [review]
I tested locally and it looks good to me.

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/ApplicationViewModel.kt [hunk 6]
```diff
@@ -112,6 +75,7 @@ class ApplicationViewModel(
         lifecycle.whileAttached {
             try {
                 syncService.runSynchronization(conference = conference)
+                syncService.forceSynchronize(conference)
```
Comment (DanielSouzaBertoldi) [inline shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/ApplicationViewModel.kt]: This seems like dead code to me 🤔

While reading the runSynchronization code, I noticed it suspends inside a coroutineScope { launch { while(isActive) } } and it just never returns while the scope is alive.

Comment (DanielSouzaBertoldi) [reply shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/ApplicationViewModel.kt]: Also, digging a bit deeper, _I believe_ that we don't even need this forced synchronization.

`runSynchronization` instantiates the `lastSessionizeSyncThisLoop` with the current time - 3 hours:

```kt
var lastSessionizeSyncThisLoop: Instant = dateTimeService.now().minus(3, DateTimeUnit.HOUR)
```

Which a bit down below `timeToSync` checks if `lastSessionizeSyncThisLoop` is equals or more than 15 minutes ago:

```kt
// simplified
val timeToSync = lastSessionizeSyncThisLoop <= dateTimeService.now().minus(15, DateTimeUnit.MINUTE)
```

Which means that `runSynchronization` will always trigger a sync when it first runs.

What do you think? Is there a corner case of some sort I'm not aware of?

File: shared/src/jsMain/kotlin/co/touchlab/droidcon/Koin.js.kt [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import co.touchlab.droidcon.domain.repository.impl.SqlDelightDriverFactory
+import co.touchlab.droidcon.service.JsNotificationService
+import co.touchlab.kermit.CommonWriter
+import co.touchlab.kermit.Logger
+import co.touchlab.kermit.StaticConfig
+import io.ktor.client.engine.HttpClientEngine
+import io.ktor.client.engine.js.Js
+import org.koin.dsl.module
+
+actual val platformModule: org.koin.core.module.Module = module {
```
Comment (DanielSouzaBertoldi) [inline shared/src/jsMain/kotlin/co/touchlab/droidcon/Koin.js.kt]:
```suggestion
actual val platformModule: Module = module {
```

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+            single<NotificationSchedulingService.LocalizedStringFactory> { NotificationLocalizedStringFactory() }
+
+            single<AnalyticsService> { WebAnalyticsService() }
+
+            single<ParseUrlViewService> { DefaultParseUrlViewService() }
+
+            // Provide JsNotificationService which is required by the JS platform module
+            single<JsNotificationService> { JsNotificationService() }
+
+            single<SqlDriver> { driver }
```
Comment (DanielSouzaBertoldi) [inline web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: Wait, I noticed that we are registering a `SqlDriver` in `Koin.js.kt` too as well as here.

So my question is: where exactly should we be registering the driver? Here in the `additionalModules` or on the platform side (`Koin.js.kt`)?

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/Koin.kt [hunk 9]
```diff
@@ -64,9 +65,9 @@ import org.koin.dsl.module
 fun initKoin(additionalModules: List<Module>): KoinApplication {
     val koinApplication = startKoin {
         modules(
-            additionalModules +
-                platformModule +
-                coreModule,
+            platformModule +
+                coreModule +
+                additionalModules,
         )
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/Koin.kt:70]: Why did we reverse the order here? Can't this affect Android/iOS too?

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/BaseSessionListViewModel.kt [hunk 10]
```diff
@@ -37,13 +38,14 @@ abstract class BaseSessionListViewModel(
...
 
+        print("Collecting Item Flows")
         itemsFlow
             .collect { items ->
                 items
                     .groupBy {
                         it.session.startsAt.toConferenceDateTime(
                             dateTimeService,
-                            conferenceConfigProvider.getConferenceTimeZone(),
+                            conferenceConfigProvider.getConferenceTimeZone() ?: kotlinx.datetime.TimeZone.UTC,
```
Comment (DanielSouzaBertoldi) [inline shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/BaseSessionListViewModel.kt]:
```suggestion
                            conferenceConfigProvider.getConferenceTimeZone() ?: TimeZone.UTC,
```

Also, same question as [here](https://github.com/touchlab/DroidconKotlin/pull/250/changes#r3318162749)

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/BaseSessionListViewModel.kt [hunk 11]
```diff
@@ -37,13 +38,14 @@ abstract class BaseSessionListViewModel(
             sessionGateway.observeSchedule()
         }
 
+        print("Collecting Item Flows")
```
Comment (DanielSouzaBertoldi) [inline shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/BaseSessionListViewModel.kt]: Is this print statement necessary?

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDayViewModel.kt [hunk 12]
```diff
@@ -27,7 +28,7 @@ class SessionDayViewModel(
             .groupBy {
                 it.session.startsAt.toConferenceDateTime(
                     dateTimeService,
-                    conferenceTimeZone = conferenceConfigProvider.getConferenceTimeZone(),
+                    conferenceTimeZone = conferenceConfigProvider.getConferenceTimeZone() ?: kotlinx.datetime.TimeZone.UTC,
```
Comment (DanielSouzaBertoldi) [inline shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDayViewModel.kt]: Hmmmm is silently falling back to UTC a good approach here? As a user, I'd rather know that the timezone changed or something like that

Just putting this out there, I know this discussion does not depend on just you and me to properly decide on a better approach 😄

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultConferenceConfigProvider.kt [hunk 13]
```diff
@@ -40,8 +41,23 @@ class DefaultConferenceConfigProvider(private val conferenceRepository: Conferen
...
+                log.w(e) { "Error observing selected conference, emitting null" }
+                emit(null)
+            }
+            .collect { conference ->
+                log.i { "loadSelectedConference: Emitting Conference! $conference" }
+                _currentConferenceState.value = conference
+            }
+    }
+
+    private suspend fun getConference(): Conference {
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultConferenceConfigProvider.kt:59]: I can see that `DefaultSyncService` is the one calling this function, and it does so like this:

```kt
val speakerDtos = dataSource.getSpeakers()
val days = dataSource.getSchedule()
val sponsorSessionsGroups = dataSource.getSponsorSessions()
```

So `getConference` here gets called at least 3 times sequentialy, since all functions here are suspend, there can be 3 concurrent calls.

In the line below, we have duplicate DB reads, one that checks `currentConference != null` and another time in `currentConference!!` (since it just gets the value from `currentConferenceState.value`).

That means we are basically paying 3 DB round-trips for the same data. I also fear we might possibly have a race condition hidden here. Given that we have two separate reads, what if one triggers the `catch` lambda above (which emits `null` to `currentConference`)  by the time another thread had just finished checking `currentConference != null`? Then `currentConference!!` will crash the app. I.e, a classic [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use).

This is very very very unlikely to happen, but better to be safe than sorry.

I think that using a Mutex here is enough to solve both issues without needing to refactor a bunch of code, but please let me know if I'm mistaken 😄 

If you do end up making such change, it'd be great to have a test for this edge case!

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 14]
```diff
@@ -20,32 +21,39 @@ class DefaultApiDataSource(
     private val json: Json,
     private val conferenceConfigProvider: ConferenceConfigProvider,
 ) : DefaultSyncService.DataSource {
-    override suspend fun getSpeakers(): List<SpeakersDto.SpeakerDto> {
+
+    private val log = Logger.withTag("DefaultApiDataSource")
+
+    override suspend fun getSpeakers(): List<SpeakersDto.SpeakerDto>? {
+        val scheduleId = conferenceConfigProvider.getScheduleId() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val scheduleId = conferenceConfigProvider.getScheduleId()
```

`getScheduleId()` can't be null

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 15]
```diff
@@ -20,32 +21,39 @@ class DefaultApiDataSource(
...
             // We want to use the same scheduleId for speakers and schedule
-            sessionize("/api/v2/${conferenceConfigProvider.getScheduleId()}/view/speakers")
+            sessionize("/api/v2/$scheduleId/view/speakers")
         }.bodyAsText()
         return json.decodeFromString(ListSerializer(SpeakersDto.SpeakerDto.serializer()), jsonString)
     }
 
-    override suspend fun getSchedule(): List<ScheduleDto.DayDto> {
+    override suspend fun getSchedule(): List<ScheduleDto.DayDto>? {
+        val scheduleId = conferenceConfigProvider.getScheduleId() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val scheduleId = conferenceConfigProvider.getScheduleId()
```

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 16]
```diff
@@ -20,32 +21,39 @@ class DefaultApiDataSource(
...
         val jsonString = client.get {
-            sessionize("/api/v2/${conferenceConfigProvider.getScheduleId()}/view/gridtable")
+            sessionize("/api/v2/$scheduleId/view/gridtable")
         }.bodyAsText()
         return json.decodeFromString(ListSerializer(ScheduleDto.DayDto.serializer()), jsonString)
     }
 
-    override suspend fun getSponsorSessions(): List<SponsorSessionsDto.SessionGroupDto> {
+    override suspend fun getSponsorSessions(): List<SponsorSessionsDto.SessionGroupDto>? {
+        val scheduleId = conferenceConfigProvider.getScheduleId() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val scheduleId = conferenceConfigProvider.getScheduleId()
```

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 17]
```diff
@@ -20,32 +21,39 @@ class DefaultApiDataSource(
...
         return json.decodeFromString(ListSerializer(SponsorSessionsDto.SessionGroupDto.serializer()), jsonString)
     }
 
-    override suspend fun getSponsors(): SponsorsDto.SponsorCollectionDto {
-        val projectId = conferenceConfigProvider.getProjectId()
-        val collectionName = conferenceConfigProvider.getCollectionName()
-        val apiKey = conferenceConfigProvider.getApiKey()
+    override suspend fun getSponsors(): SponsorsDto.SponsorCollectionDto? {
+        log.i { "gettingSponsors" }
+        val projectId = conferenceConfigProvider.getProjectId() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val projectId = conferenceConfigProvider.getProjectId()
```

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 18]
```diff
@@ -20,32 +21,39 @@ class DefaultApiDataSource(
...
+        val collectionName = conferenceConfigProvider.getCollectionName() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val collectionName = conferenceConfigProvider.getCollectionName()
```

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 19]
```diff
@@ -20,32 +21,39 @@ class DefaultApiDataSource(
...
+        val apiKey = conferenceConfigProvider.getApiKey() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val apiKey = conferenceConfigProvider.getApiKey()
```

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 20]
```diff
@@ -54,9 +62,12 @@ class DefaultApiDataSource(
         return json.decodeFromString(SponsorsDto.SponsorCollectionDto.serializer(), jsonString)
     }
 
-    suspend fun getConferences(): ConferencesDto.ConferenceCollectionDto {
-        val projectId = conferenceConfigProvider.getProjectId()
-        val apiKey = conferenceConfigProvider.getApiKey()
+    suspend fun getConferences(): ConferencesDto.ConferenceCollectionDto? {
+        log.i { "getConferences" }
+        val projectId = conferenceConfigProvider.getProjectId() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val projectId = conferenceConfigProvider.getProjectId()
```

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 21]
```diff
@@ -54,9 +62,12 @@ class DefaultApiDataSource(
...
+        log.i { "Have Project ID: $projectId" }
+        val apiKey = conferenceConfigProvider.getApiKey() ?: return null
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]:
```suggestion
        val apiKey = conferenceConfigProvider.getApiKey()
```

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt [hunk 22]
```diff
@@ -54,9 +62,12 @@ class DefaultApiDataSource(
         return json.decodeFromString(SponsorsDto.SponsorCollectionDto.serializer(), jsonString)
     }
 
-    suspend fun getConferences(): ConferencesDto.ConferenceCollectionDto {
-        val projectId = conferenceConfigProvider.getProjectId()
-        val apiKey = conferenceConfigProvider.getApiKey()
+    suspend fun getConferences(): ConferencesDto.ConferenceCollectionDto? {
+        log.i { "getConferences" }
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]: Also, do we need this many logs? I feel like it's a bit excessive. Wdyt?

File: web/src/wasmJsMain/kotlin/org/example/project/Platform.wasmJs.kt [hunk 23]
```diff
@@ -0,0 +1,7 @@
+package org.example.project
+
+class WasmPlatform : Platform {
+    override val name: String = "Web with Kotlin/Wasm"
+}
+
+actual fun getPlatform(): Platform = WasmPlatform()
```
Comment (DanielSouzaBertoldi) [inline web/src/wasmJsMain/kotlin/org/example/project/Platform.wasmJs.kt]: Since we can't support wasm for now, does it make sense to have this placeholder here right now?

File: shared/build.gradle.kts [hunk 24]
```diff
@@ -61,35 +62,71 @@ kotlin {
...
+    js(IR) {
+        browser()
+        binaries.executable()
+    }
 
     version = "1.0"
 
     sourceSets {
         commonMain.dependencies {
+            // Required when composeCompiler plugin is applied in this module.
```
Comment (DanielSouzaBertoldi) [inline shared/build.gradle.kts:74]: I'm not familiar with Composable resources for KMP, so I have to ask, why does  `composeCompiler` needs to be on `:shared` at all? To me it makes more sense to have it only on `shared-ui`

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/ui/MainComposeView.kt [hunk 25]
```diff
@@ -85,6 +55,7 @@ private fun MainAppBody(waitForLoadedContextModel: WaitForLoadedContextModel, se
             viewModel.selectedTab = ApplicationViewModel.Tab.Schedule
         }
 
+        print("isFirstRun")
```
Comment (DanielSouzaBertoldi) [inline shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/ui/MainComposeView.kt]:
```suggestion
```

File: web/src/jsMain/kotlin/co/touchlab/droidcon/web/Main.kt [hunk 26]
```diff
@@ -0,0 +1,45 @@
+package co.touchlab.droidcon.web
```
Comment (DanielSouzaBertoldi) [inline web/src/jsMain/kotlin/co/touchlab/droidcon/web/Main.kt:1]: The name of this file is `Main2.kt`. Could we rename it to just `Main.kt`?

File: shared-ui/build.gradle.kts [hunk 27]
```diff
@@ -9,6 +9,11 @@ plugins {
     alias(libs.plugins.composeCompiler)
 }
 
+// Force Kotlin stdlib for JS so compiler sees stdlib (fixes "Symbol for Any not found")
```
Comment (DanielSouzaBertoldi) [inline shared-ui/build.gradle.kts:12]: 😯 that's interesting. Maybe the kotlin version bump surfaced an inconsistency in the dependency tree?

File: web/build.gradle.kts [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+                implementation(compose.material)
+                implementation(libs.multiplatformSettings.core)
+                implementation(libs.multiplatform.settings.make.observable)
+                implementation(libs.koin.compose)
+                implementation(libs.hyperdrive.multiplatformx.api)
+            }
+        }
+        val jsTest by getting {
+            dependencies {
+                implementation("org.jetbrains.kotlin:kotlin-test:2.2.21")
```
Comment (DanielSouzaBertoldi) [inline web/build.gradle.kts]:
```suggestion
                implementation(libs.kotlin.test)
```

Better to reference dependencies in the TOML file here too 😄

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+            startsAtAdapter = InstantSqlDelightAdapter,
+            endsAtAdapter = InstantSqlDelightAdapter,
+            feedbackRatingAdapter = intToLongAdapter,
+        ),
+        sponsorGroupTableAdapter = SponsorGroupTable.Adapter(
+            intToLongAdapter,
+        ),
+        conferenceTableAdapter = ConferenceTable.Adapter(
+            conferenceTimeZoneAdapter = timeZoneAdapter,
+            // Note: selectedAdapter will be added when the adapter is regenerated
```
Comment (DanielSouzaBertoldi) [inline web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: When exactly is the adapter going to be regenerated? Without it the conference table might not deserialize correctly on web right?

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/util/formatter/KotlinXDateFormatter.kt [hunk 30]
```diff
@@ -0,0 +1,33 @@
+package co.touchlab.droidcon.util.formatter
+
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+
+/**
+ * Shared [DateFormatter] for all platforms using kotlinx-datetime only (no java.text / NSDateFormatter / JS stubs).
+ * Schedule data uses conference-local [LocalDate] / [LocalDateTime]; we format those fields directly.
+ */
+class KotlinXDateFormatter : DateFormatter {
```
Comment (DanielSouzaBertoldi) [inline shared/src/commonMain/kotlin/co/touchlab/droidcon/util/formatter/KotlinXDateFormatter.kt:10]: I see that there's no test class for this class. Could we add some please? 😄

File: web/src/jsTest/kotlin/co/touchlab/droidcon/web/WebKoinTest.kt [hunk 31]
```diff
@@ -0,0 +1,54 @@
...
+import org.koin.dsl.koinApplication
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+import org.koin.test.check.checkModules
+
+class WebKoinTest : KoinTest {
+
+    @OptIn(ExperimentalSettingsApi::class)
+    @Test
+    fun `should inject my components`() {
```
Comment (DanielSouzaBertoldi) [inline web/src/jsTest/kotlin/co/touchlab/droidcon/web/WebKoinTest.kt]: This tests the DI graph, but there's no tests that verifies that the app actually initialized correctly

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDayViewModel.kt (hunk 12 above)
Comment (DanielSouzaBertoldi) [inline shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDayViewModel.kt]:
```suggestion
                    conferenceTimeZone = conferenceConfigProvider.getConferenceTimeZone() ?: TimeZone.UTC,
```

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDetailViewModel.kt [hunk 32]
```diff
@@ -57,9 +58,10 @@ class SessionDetailViewModel(
         listOfNotNull(
             it.room?.name,
             with(dateTimeService) {
+                val timeZone = conferenceConfigProvider.getConferenceTimeZone() ?: kotlinx.datetime.TimeZone.UTC
```
Comment (DanielSouzaBertoldi) [inline shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDetailViewModel.kt]: Same question as [here](https://github.com/touchlab/DroidconKotlin/pull/250/changes#r3318162749)


```suggestion
                val timeZone = conferenceConfigProvider.getConferenceTimeZone() ?: TimeZone.UTC
```

**Review by DanielSouzaBertoldi — CHANGES_REQUESTED** [review]
Good direction overall, most of the comments are really minor stuff, but there are some of them I think it's really important that we address before we merge. Great work! 🚀

**Review by DanielSouzaBertoldi — CHANGES_REQUESTED** [review]
Good direction overall, most of the comments are really minor stuff, but there are some of them I think it's really important that we address before we merge. Great work! 🚀

File: web/src/jsMain/kotlin/co/touchlab/droidcon/web/Main.kt (hunk 26 above)
Comment (KevinSchildhorn) [reply web/src/jsMain/kotlin/co/touchlab/droidcon/web/Main.kt:1]: Updated

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt (hunk 8 above)
Comment (KevinSchildhorn) [reply web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: I updated it to only be called in the `DependencyInjection` file as it needs a suspend call for this version.

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDayViewModel.kt (hunk 12 above)
Comment (KevinSchildhorn) [reply shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/SessionDayViewModel.kt]: Yea that's a fair point. I know that the main reason we had to change it to optional was a race condition so it might be alright for the most part. Maybe instead I'll default it to the default TimeZone of the Conference

File: web/build.gradle.kts (hunk 28 above)
Comment (KevinSchildhorn) [reply web/build.gradle.kts]: Updated

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt (hunk 22 above)
Comment (KevinSchildhorn) [reply shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultApiDataSource.kt]: Yea that was probably from a test. I'll remove these

File: web/src/jsTest/kotlin/co/touchlab/droidcon/web/WebKoinTest.kt (hunk 31 above)
Comment (KevinSchildhorn) [reply web/src/jsTest/kotlin/co/touchlab/droidcon/web/WebKoinTest.kt]: I'm going to remove this because it doesn't really help at this point. That was during testing.

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/util/formatter/KotlinXDateFormatter.kt (hunk 30 above)
Comment (KevinSchildhorn) [reply shared/src/commonMain/kotlin/co/touchlab/droidcon/util/formatter/KotlinXDateFormatter.kt:10]: Added a test

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt (hunk 29 above)
Comment (KevinSchildhorn) [reply web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: I'm not sure. It's an object so maybe it won't? I'm not sure exactly what you mean sorry

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/Koin.kt (hunk 9 above)
Comment (KevinSchildhorn) [reply shared/src/commonMain/kotlin/co/touchlab/droidcon/Koin.kt:70]: I'll revert this and see if it works. It might've been based on some of the SQL koin work I'm reverting.

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/BaseSessionListViewModel.kt (hunk 11 above)
Comment (KevinSchildhorn) [reply shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/session/BaseSessionListViewModel.kt]: removed

File: shared-ui/build.gradle.kts (hunk 27 above)
Comment (KevinSchildhorn) [reply shared-ui/build.gradle.kts:12]: Yea I don't quite remember. I think this one might've been an ai fix for a certain issue.

File: shared/build.gradle.kts (hunk 24 above)
Comment (KevinSchildhorn) [reply shared/build.gradle.kts:74]: If I remember correctly it was because we have an `about.json` in composeResources. It wasn't a UI thing it was a file issue.

File: web/src/wasmJsMain/kotlin/org/example/project/Platform.wasmJs.kt (hunk 23 above)
Comment (KevinSchildhorn) [reply web/src/wasmJsMain/kotlin/org/example/project/Platform.wasmJs.kt]: Good point. I've removed this file.

File: shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/ApplicationViewModel.kt (hunk 6 above)
Comment (KevinSchildhorn) [reply shared-ui/src/commonMain/kotlin/co.touchlab.droidcon/viewmodel/ApplicationViewModel.kt]: I think there was a corner case but I don't remember at this time. Let me play around with it and see if there's an issue.

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt (hunk 29 above)
Comment (DanielSouzaBertoldi) [reply web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: I was asking because of the comment "selectedAdapter will be added when the adapter is regenerated", which to me looks like some leftover work? Idk, I'm not sure either what it means lol

File: shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultConferenceConfigProvider.kt (hunk 13 above)
Comment (KevinSchildhorn) [reply shared/src/commonMain/kotlin/co/touchlab/droidcon/domain/service/impl/DefaultConferenceConfigProvider.kt:59]: I've added a mutex for this, take a look

File: web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt (hunk 29 above)
Comment (KevinSchildhorn) [reply web/src/webMain/kotlin/co/touchlab/droidcon/web/DependencyInjection.kt]: Yea it was a leftover, sorry.

**Review by DanielSouzaBertoldi — APPROVED** [review]
